data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/CneCnecsCreator.java

```diff
@@ -0,0 +1,174 @@
+        CnecResultExtension cnecResultExtension = cnec.getExtension(CnecResultExtension.class);
+        assert cnecResultExtension != null;
+
+        CnecResult cnecResultPost = cnecResultExtension.getVariant(postOptimVariantId);
+        if (cnecResultPost != null) {
+            // Flow and threshold in A
+            addFlowThreshold(cnecResultPost, Unit.AMPERE, measurementType, measurementsB54, measurementsB57);
+            // Flow and threshold in MW
+            addFlowThreshold(cnecResultPost, Unit.MEGAWATT, measurementType, measurementsB54, measurementsB57);
+
+            // sumPTDF
+            addSumPtdf();
+
+            // loopflow
+            addLoopflow(cnecResultPost, measurementsB54);
+            addLoopflow(cnecResultPost, measurementsB57);
+        }
+    }
+
+    // B88
+    public static void createB88Measurements(Cnec cnec, String measurementType, String preOptimVariantId, List<Analog> measurementsB88) {
+        CnecResultExtension cnecResultExtension = cnec.getExtension(CnecResultExtension.class);
+        // The check of the existence of the CnecResultExtension was done in another method
+        assert cnecResultExtension != null;
+
+        CnecResult cnecResultPre = cnecResultExtension.getVariant(preOptimVariantId);
+        if (cnecResultPre != null) {
+            // Flow and threshold in A
+            addFlowThreshold(cnecResultPre, Unit.AMPERE, measurementType, measurementsB88);
+            // Flow and threshold in MW
+            addFlowThreshold(cnecResultPre, Unit.MEGAWATT, measurementType, measurementsB88);
+            // FRM
+            addFrm(cnec, measurementsB88);
+
+            // sumPTDF
+            addSumPtdf();
+
+            // loopflow
+            addLoopflow(cnecResultPre, measurementsB88);
+        }
+    }
+
+    // Flow and threshold
+    private static void addFlowThreshold(CnecResult cnecResult, Unit unit, String measurementType, List<Analog> measurements) {
+        addFlowThreshold(cnecResult, unit, measurementType, measurements, measurements, true);
+    }
+
+    private static void addFlowThreshold(CnecResult cnecResult, Unit unit, String measurementType, List<Analog> measurementsPatl, List<Analog> measurementsTatl) {
+        addFlowThreshold(cnecResult, unit, measurementType, measurementsPatl, measurementsTatl, false);
+    }
+
+    private static void addFlowThreshold(CnecResult cnecResult, Unit unit, String measurementType, List<Analog> measurementsPatl, List<Analog> measurementsTatl, boolean b88) {
+
+        double flow;
+        double threshold;
+        // cnecResult is not null, checked before
+        assert cnecResult != null;
+        if (unit.equals(Unit.AMPERE)) {
+            flow = cnecResult.getFlowInA();
+            if (flow < 0) {
+                threshold = cnecResult.getMinThresholdInA();
+            } else {
+                threshold = cnecResult.getMaxThresholdInA();
```

Comment:
I am not sure if this check is relevant or not... I would have put cnecResult.getMaxThresholdInA in all cases because I am not sure how MinThreshold should be handled in the CBCORA/CNE (if they are supported).

Reply:
Are you sure that what you propose would be more reliable?
@baptsegu What do you think?